compiler/rustc_data_structures/src/graph/vec_graph/mod.rs

```diff
@@ -20,7 +20,7 @@ pub struct VecGraph<N: Idx> {
 impl<N: Idx> VecGraph<N> {
     pub fn new(num_nodes: usize, mut edge_pairs: Vec<(N, N)>) -> Self {
         // Sort the edges by the source -- this is important.
-        edge_pairs.sort();
+        edge_pairs.sort_by_cached_key(|&edge_pairs| (edge_pairs.0.index(), edge_pairs.1.index()));
```

Comment:
Why use the `cached_` version?

Reply:
No particular reason. I'd seen this method used over `sort_by_key` elsewhere in the compiler.